packages/sources/intrinio-test/src/endpoint/price-ws.ts

```diff
@@ -0,0 +1,75 @@
+import { WebSocketTransport } from '@chainlink/external-adapter-framework/transports'
+import { SingleNumberResultResponse } from '@chainlink/external-adapter-framework/util'
+import { customSettings } from '../config'
+import { IntrinioRealtime } from './util'
+
+export type IntrinioFeedMessage = {
+  topic: string
+  payload: {
+    type: string
+    timestamp: number
+    ticker: string
+    size: number
+    price: number
+  }
+  event: string
+}
+
+export type EndpointTypes = {
+  Request: {
+    Params: { base: string }
+  }
+  Response: SingleNumberResultResponse
+  CustomSettings: typeof customSettings
+  Provider: {
+    WsMessage: IntrinioFeedMessage[]
+  }
+}
+
+let ws: IntrinioRealtime
+export const wsTransport = new WebSocketTransport<EndpointTypes>({
+  url: (context) => {
+    const { API_KEY } = context.adapterConfig
+    if (!ws) {
+      ws = new IntrinioRealtime({
+        api_key: API_KEY,
+        provider: 'iex',
+      })
+    }
+    return ws._makeSocketUrl.bind(ws)()
+  },
```

Comment:
Same as the conversation on the tiingo-test EA, we should consider creating a new class instead of holding global variables: https://github.com/smartcontractkit/external-adapters-js/pull/2430#discussion_r1086664430

Reply:
Done